README.md

```diff
@@ -145,6 +145,66 @@ final class CompanyName
 }
 ```
 
+#### Tool Return Value
+
+In the end, the tool's response needs to be a string, but LLM Chain converts arrays and objects, that implement the
+`JsonSerializable` interface, to JSON strings for you. So you can return arrays or objects directly from your tool.
+
+#### Tool Methods
+
+You can configure the method to be called by the LLM with the `#[AsTool]` attribute and have multiple tools per class:
+
+```php
+use PhpLlm\LlmChain\ToolBox\Attribute\AsTool;
+
+
+#[AsTool(name: 'weather_current', description: 'get current weather for a location', method: 'current')]
+#[AsTool(name: 'weather_forecast', description: 'get weather forecast for a location', method: 'forecast')]
+final readonly class OpenMeteo
+{
+    public function current(float $latitude, float $longitude): array
+    {
+        // ...
+    }
+
+    public function forecast(float $latitude, float $longitude): array
+    {
+        // ...
+    }
+}
+```
+
+#### Tool Parameters
+
+LLM Chain generates a JSON Schema representation for all tools in the `ToolBox` based on the `#[AsTool]` attribute and
+method arguments and param comments in the doc block. Additionally, JSON Schema support validation rules, which are
+partially support by LLMs like GPT.
+
+To leverage this, configure the `#[ToolParameter]` attribute on the method arguments of your tool:
```

Comment:
Maybe add a link to the class ToolParameter? 

Offtopic, but maybe it's more fail safe to have sth like "#[Minimum]" etc as we can then validate it is only used for integers for example 🤔 to avoid god classes

Reply:
added a comment below the example code to link to the attribute class, good idea :+1: 

i do get the intention in resolving the mega attribute class, but on the other hand it would make it harder to parse that thing in `ParameterAnalyzer`